Lint issues with autogenerated code from lxd-generate

<!--
Github issues are used for bug reports. For support questions, please use [our forum](https://discuss.linuxcontainers.org).

Please fill the template below as it will greatly help us track down your issue and reproduce it on our side.  
Feel free to remove anything which doesn't apply to you and add more information where it makes sense.
-->

# Required information

 * Distribution: Ubuntu
 * Distribution version: 22.04

# Issue description

Project https://github.com/openstack-snaps/sunbeam-microcluster uses lxd-generate to autogenerate database mapper files.
However ``make check`` has lint issues with respect to autogenerated code

```
$ make check
CGO_LDFLAGS_ALLOW="-Wl,-z,now" golangci-lint run --timeout 5m
revive -set_exit_status ./...
database/config.mapper.go:273:23: parameter 'ctx' seems to be unused, consider removing or renaming it as _
database/jujuuser.mapper.go:273:21: parameter 'ctx' seems to be unused, consider removing or renaming it as _
database/node.mapper.go:373:17: parameter 'ctx' seems to be unused, consider removing or renaming it as _
make: *** [Makefile:30: check-static] Error 1
```

This is specifically in Delete<> function generated by ``//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e ConfigItem DeleteOne-by-Key table=<>``.
 
Following has fixed the issue
```
$ git diff database/config.mapper.go
diff --git a/database/config.mapper.go b/database/config.mapper.go
index 88e37a9..ba47988 100644
--- a/database/config.mapper.go
+++ b/database/config.mapper.go
@@ -270,7 +270,7 @@ func CreateConfigItem(ctx context.Context, tx *sql.Tx, object ConfigItem) (int64
 
 // DeleteConfigItem deletes the ConfigItem matching the given key parameters.
 // generator: ConfigItem DeleteOne-by-Key
-func DeleteConfigItem(ctx context.Context, tx *sql.Tx, key string) error {
+func DeleteConfigItem(_ context.Context, tx *sql.Tx, key string) error {
        stmt, err := cluster.Stmt(tx, configItemDeleteByKey)
        if err != nil {
                return fmt.Errorf("Failed to get \"configItemDeleteByKey\" prepared statement: %w", err)
```

# Steps to reproduce

 1. git clone https://github.com/openstack-snaps/sunbeam-microcluster.git
 2. CGO_LDFLAGS_ALLOW="-Wl,-z,now" make check

--- a/lxd/db/cluster/config.mapper.go
+++ b/lxd/db/cluster/config.mapper.go
@@ -158,7 +158,7 @@ func CreateConfig(ctx context.Context, tx *sql.Tx, parent string, object Config)
 	}
 
 	queryStr := fmt.Sprintf(configCreateLocal, fillParent...)
-	_, err := tx.Exec(queryStr, object.ReferenceID, object.Key, object.Value)
+	_, err := tx.ExecContext(ctx, queryStr, object.ReferenceID, object.Key, object.Value)
 	if err != nil {
 		return fmt.Errorf("Insert failed for \"%s_config\" table: %w", parent, err)
 	}
@@ -202,7 +202,7 @@ func DeleteConfig(ctx context.Context, tx *sql.Tx, parent string, referenceID in
 	}
 
 	queryStr := fmt.Sprintf(configDeleteLocal, fillParent...)
-	result, err := tx.Exec(queryStr, referenceID)
+	result, err := tx.ExecContext(ctx, queryStr, referenceID)
 	if err != nil {
 		return fmt.Errorf("Delete entry for \"%s_config\" failed: %w", parent, err)
 	}

--- a/lxd/db/cluster/devices.mapper.go
+++ b/lxd/db/cluster/devices.mapper.go
@@ -180,7 +180,7 @@ func CreateDevices(ctx context.Context, tx *sql.Tx, parent string, objects map[s
 
 	queryStr := fmt.Sprintf(deviceCreateLocal, fillParent...)
 	for _, object := range objects {
-		result, err := tx.Exec(queryStr, object.ReferenceID, object.Name, object.Type)
+		result, err := tx.ExecContext(ctx, queryStr, object.ReferenceID, object.Name, object.Type)
 		if err != nil {
 			return fmt.Errorf("Insert failed for \"%s_devices\" table: %w", parent, err)
 		}
@@ -241,7 +241,7 @@ func DeleteDevices(ctx context.Context, tx *sql.Tx, parent string, referenceID i
 	}
 
 	queryStr := fmt.Sprintf(deviceDeleteLocal, fillParent...)
-	result, err := tx.Exec(queryStr, referenceID)
+	result, err := tx.ExecContext(ctx, queryStr, referenceID)
 	if err != nil {
 		return fmt.Errorf("Delete entry for \"%s_device\" failed: %w", parent, err)
 	}

--- a/lxd/db/generate/db/method.go
+++ b/lxd/db/generate/db/method.go
@@ -704,10 +704,10 @@ func (m *Method) create(buf *file.Buffer, replace bool) error {
 
 		refFields := mapping.RefFields()
 		if len(refFields) == 0 {
-			buf.L("_, err := tx.Exec(queryStr, %s)", createParams)
+			buf.L("_, err := tx.ExecContext(ctx, queryStr, %s)", createParams)
 			m.ifErrNotNil(buf, true, fmt.Sprintf(`fmt.Errorf("Insert failed for \"%%s_%s\" table: %%w", parent, err)`, lex.Plural(m.entity)))
 		} else {
-			buf.L("result, err := tx.Exec(queryStr, %s)", createParams)
+			buf.L("result, err := tx.ExecContext(ctx, queryStr, %s)", createParams)
 			m.ifErrNotNil(buf, true, fmt.Sprintf(`fmt.Errorf("Insert failed for \"%%s_%s\" table: %%w", parent, err)`, lex.Plural(m.entity)))
 			buf.L("id, err := result.LastInsertId()")
 			m.ifErrNotNil(buf, true, "fmt.Errorf(\"Failed to fetch ID: %w\", err)")
@@ -1119,7 +1119,7 @@ func (m *Method) delete(buf *file.Buffer, deleteOne bool) error {
 		buf.L("}")
 		buf.N()
 		buf.L("queryStr := fmt.Sprintf(%s, fillParent...)", stmtLocal)
-		buf.L("result, err := tx.Exec(queryStr, referenceID)")
+		buf.L("result, err := tx.ExecContext(ctx, queryStr, referenceID)")
 		m.ifErrNotNil(buf, true, fmt.Sprintf(`fmt.Errorf("Delete entry for \"%%s_%s\" failed: %%w", parent, err)`, m.entity))
 	} else {
 		activeFilters := mapping.ActiveFilters(m.kind)

